src/astra_driver.cpp

```diff
@@ -199,6 +203,18 @@ bool AstraDriver::EnableStreaming(std_srvs::SetBool::Request &req, std_srvs::Set
   return true;
 }
 
+void AstraDriver::setHealthTimers() {
+  auto reset_this = [this](const ros::TimerEvent&) -> void
+  {
+    ROS_WARN_STREAM("Astra " << ns_ << " driver timeout! Resetting");
```

Comment:
This timer callback happens 30s after device_->startDepthStream(); from both line 186 and 544. Given this, shouldn't we check whether or not stream is ok before we decide to destruct and reconstruct the object?

Reply:
If the stream is ok, the depth callback will reset the timer  and the timer callback won't happen at all